lib/puppet/functions/gitlab_ci_runner/register_to_file.rb

```diff
@@ -25,17 +26,19 @@
     param 'String[1]', :regtoken
     param 'String[1]', :runner_name
     optional_param 'Hash', :additional_options
-    optional_param 'String[1]', :filename
+    optional_param 'Optional[String[1]]', :filename
+    optional_param 'Optional[String[1]]', :proxy
     return_type 'String[1]'
   end
 
-  def register_to_file(url, regtoken, runner_name, additional_options = {}, filename = "/etc/gitlab-runner/auth-token-#{runner_name}")
+  def register_to_file(url, regtoken, runner_name, additional_options = {}, filename = nil, proxy = nil)
```

Comment:
I'm wondering if `filename` really needs to be a parameter??

Reply:
No, I think I only added it for debugging/convenience. Do you want to remove it?

Reply:
sure.  It'd make the code a little cleaner.

Reply:
updated